nodes/usb_cam_node.cpp

```diff
@@ -247,21 +254,23 @@ class UsbCamNode
     ros::Rate loop_rate(this->framerate_);
     while (node_.ok())
     {
-      if (cam_.is_capturing()) {
-        if (!take_and_send_image()) ROS_WARN("USB camera did not respond in time.");
+      if (cam_.is_capturing()) { 
+        if (!take_and_send_image())
+          ROS_WARN("USB camera did not respond in time.");
+      } else if (camera_is_connected()) {
+        this->init_camera();
```

Comment:
is there anyway the 'init' call can fail?

Reply:
I'm guessing the original authors weren't worried about needing to stop and restart the camera in the event of the device failing for whatever reason, so all the code in 'init_camera' was originally inside the constructor. I just pulled that chunk out so it could be called as needed. As far as I can tell, so long as the first call to 'init_camera' in the constructor succeeds, any subsequent calls to restart the camera should succeed since its using the same data. If it does fail, its handled however the original authors wrote it, which looks to be by calling node.shutdown().

Reply:
So this standard thing to do if something fails in a ctor is to throw and exception so that the object cannot be created. Otherwise, you get left in some half-constructed state. Would it be straight forward to check the return for `init` in the ctor, and raise an exception if it fails? Then just have the calling code try to construct the object again?

Reply:
Sure, I can change `init_camera` to return a boolean, and then just check that on the call in the constructor. Is a custom exception, say `CameraInitializationException` necessary, or just use a more straight-forward, general exception?